x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/index.ts

```diff
@@ -576,10 +426,60 @@ export class KnowledgeBaseService {
 
       return Promise.resolve();
     } catch (error) {
-      if (isModelMissingOrUnavailableError(error)) {
+      if (isInferenceEndpointMissingOrUnavailable(error)) {
         throwKnowledgeBaseNotReady(error.body);
       }
       throw error;
     }
   };
+
+  getStatus = async () => {
+    let errorMessage = '';
+    const endpoint = await getInferenceEndpoint({
+      esClient: this.dependencies.esClient,
+      logger: this.dependencies.logger,
+    }).catch((error) => {
+      if (!isInferenceEndpointMissingOrUnavailable(error)) {
+        throw error;
+      }
+      this.dependencies.logger.error(`Failed to get inference endpoint: ${error.message}`);
+      errorMessage = error.message;
+    });
+
+    const enabled = this.dependencies.enabled;
+    if (!endpoint) {
+      return { ready: false, enabled, errorMessage };
+    }
+
+    const modelId = endpoint.service_settings?.model_id;
+    const modelStats = await this.dependencies.esClient.asInternalUser.ml
+      .getTrainedModelsStats({ model_id: modelId })
+      .catch((error) => {
+        this.dependencies.logger.error(`Failed to get model stats: ${error.message}`);
+        errorMessage = error.message;
+      });
+
+    if (!modelStats) {
+      return { ready: false, enabled, errorMessage };
+    }
+
+    const elserModelStats = modelStats.trained_model_stats[0];
```

Comment:
This is a nasty issue but you don't want the first endpoint returned, you want the one that has a deployment_id matching your inference endpoint name (in this case the constant you defined earlier). The semantic text endpoint only interfaces with the specific deployment of the model with the inference endpoint's name, not any other deployment of the same model id.

Reply:
Thanks! I blindly copied what we had already but what you say makes sense. 

```suggestion
  const elserModelStats = modelStats.trained_model_stats.find(
      (stats) => stats.model_id === modelId
    );
```

Btw This is only used for debugging information to the user in case anything is not working as expected.

Reply:
Oh, you said `deployment_id`, not `model_id`. So rather like this?

```
const elserModelStats = modelStats.trained_model_stats.find(
      (stats) => stats.deployment_stats?.deployment_id === AI_ASSISTANT_KB_INFERENCE_ID
    );
```

Reply:
Oh, you said `deployment_id`, not `model_id`. So rather like this?

```suggestion
    const elserModelStats = modelStats.trained_model_stats.find(
      (stats) => stats.deployment_stats?.deployment_id === AI_ASSISTANT_KB_INFERENCE_ID
    );
```

Reply:
Yes, exactly!